kraft/cmd/push.py

```diff
@@ -0,0 +1,198 @@
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+from opencontainers.distribution.reggie import *
+from kraft.logger import logger
+import click
+import sys
+import os
+import tarfile
+import json
+
+default_path = './package/'
+index_path = 'index.json'
+default_url = 'http://10.16.4.30:80'
+default_project = 'library'
+default_user = 'admin'
+default_passwd = 'Harbor12345'
+
+# TODO: add version tag for repositories
+@click.command('push', short_help='Push an OCI-Image on Harbor')
+
+@click.option(
+    '--image', '-i', 'image',
+    help='Specify the OCI-Image path. Default: ./package/*',
+    metavar="IMAGE_PATH"
+)
+
+@click.option(
+    '--name', '-n', 'name',
+    help='Specify the repository name. Default: IMAGE_NAME',
+    metavar="REPOSITORY"
+)
+
+@click.option(
+    '--server', '-s', 'server',
+    help='Specify a Harbor instance url',
+    metavar="http://URL:PORT",
+    default=default_url
+)
+
+@click.option(
+    '--project', '-p', 'project',
+    help='Specify the project',
+    metavar="PROJECT",
+    default=default_project
+)
+
+@click.option(
+    '--user', '-u', 'user',
+    help='Specify the user',
+    metavar="USER",
+    default=default_user
+)
+
+@click.option(
+    '--password', '-pw', 'passwd',
+    help='Specify the password',
+    metavar="PASSWORD",
+    default=default_passwd
+)
+
+@click.pass_context
+def cmd_push(ctx, image=None, name=None, server=None, project=None, user=None, passwd=None):
+    """
+    Push an OCI-Image on Harbor
+    """
+
+    if image is None:
+        files = os.listdir(default_path)
+        if len(files) != 1:
+            logger.critical('Operation failed, none or multiple files found.')
+            if ctx.obj.verbose:
+                import traceback
+                logger.critical(traceback.format_exc())
+            sys.exit(1)
+        else:
+            image = default_path + files[0]
+
+    if name is None:
+        tokens = image.split('/')
+        name = tokens[len(tokens)-1]
+
+    try:
+        kraft_push(
+            image=image,
+            name=name,
+            server=server,
+            project=project,
+            user=user,
+            passwd=passwd
+        )
+    except Exception as e:
+        logger.critical(str(e))
+
+        if ctx.obj.verbose:
+            import traceback
+            logger.critical(traceback.format_exc())
+
+        sys.exit(1)
+
+@click.pass_context
+def kraft_push(ctx, image=None, name=None, server=None, project=None, user=None, passwd=None):
+  
+    if not os.path.isfile(image):
+        raise Exception('Image not found.')
+
+    rc = tarfile.is_tarfile(image)
+    if not rc:
+        raise Exception('Bad OCI-Image format.')
```

Comment:
Perhaps something a bit more descriptive could be helpful, e.g. archive is not a tar or oci-image format.